src/builders.ts

```diff
@@ -149,16 +164,20 @@ export async function broadcastRawTransaction(rawTx: Buffer, url: string) {
 
   const response = await fetchPrivate(url, options);
   if (!response.ok) {
-    let msg = '';
     try {
-      msg = await response.text();
-    } catch (error) {}
-    throw new Error(
-      `Error broadcasting transaction. Response ${response.status}: ${response.statusText} fetching ${url} - ${msg}`
-    );
+      const responseData: TxBroadcastErrorResponseData = await response.json();
+      return {
```

Comment:
Should we be essentially returning errors in 2 different ways here? For a consumer of this function, you would need to both handle errors in a `response.ok = false` branch and in the `catch` block. IMO, ideally they could all be handled by the `catch` block.

Reply:
I think there's a distinction to be made between an "Broadcast Error" (meaning the broadcast failed because the stacks node returned an error), and an error broadcasting (the node couldn't be reached, etc). While I agree that it's confusing to account for both kinds of errors here, I'd like to keep `TxBroadcastError` restricted to errors that occur on the stacks node... How do you suggest I simplify this without expanding the meaning of `TxBroadcastError`?

Reply:
Seems to me like error reaching the node would fall into `TxBroadcastError`.

Reply:
The actual type that is included in the response is `TxBroadcastResultError`, which contains a `TxBroadcastError` (perhaps this should be renamed) value (this is an enum corresponding to the error types in the Rust codebase). I think this naming makes it relatively clear that the broadcast happened, and that there was an error on the stacks node.

I still think there should be two kinds of errors here, since they will likely be handled differently. Do you think that we should throw an Error (I'd make a custom error class here) instead of returning a Response object if there's an error on the stacks node? I think it's kinda nice that this function explicitly returns a result type (I wish this is how all error handling worked in JS), but I agree that it's confusing that there's two kinds of "errors" that can be returned/thrown...

Reply:
It'd be nice to throw the `TxBroadcastResultError` as an error for the consumer. And throw network errors as well.